### ISSUE TITLE
Bug Report: Agent picker won't pick matching name containing apostrophe

### ISSUE THREAD

**Issue by AJLinn** [ArctosDB/arctos#8175]
**Describe the bug**
I created an agent for the local business "Santa's Stitches", added a profile and phone number to the agent record.
I then tried to assign the exact agent to the creator field. 
The agent picker gave me a pop-up with every agent that included the string "Santa" instead of auto-populating the field with the exact agent.

<img width="1413" alt="Screenshot 2024-10-07 at 3 53 01 PM" src="https://github.com/user-attachments/assets/a83cde77-7c5f-4e44-98eb-41d49def0202">

**Data**
Catalog record https://arctos.database.museum/guid/UAM:EH:UA2024-008-0021
Agent record https://arctos.database.museum/edit_agent.cfm?agent_id=21356197

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Mac OS 14.6.1
 - Browser [e.g. chrome, safari] Firefox
 - Version [e.g. 22] 128.3.0esr (64 bit)

**Additional context**
I hadn't verified the agent entry yet, if that makes a difference?

**Comment by mkoo** [comment]
HI Angie, this has happened to me. After the alarming pick cell, try selecting the agent Santa's Stitches anyway-- should convert the cell green?! I just tried that myself with the same agent and worked. please confirm it works for you and close the issue!

**Comment by dustymc** [comment]
This is not a bug, the auto-select is purposefully limited to "simple" agents (to avoid any pitfalls in the automation). I could possibly be a bit less choosy with what's available to automation if that's really critical, but I think there will always be some limitations.

FWIW it's still pretty wild to me that any agent ever gets used without being carefully chosen. We type "James Smith" and if there's one in Arctos we just assume that's definitely who we want and couldn't possibly be one of the thirty thousand or so people who are sharing that name right this minute....

**Comment by AJLinn** [comment]
> confirm it works for you and close the issue!

Yes, it worked but I had to sift thru about 32 agents who are associated with Santa Barbara or have "Santa" as part of their name string.

> the auto-select is purposefully limited to "simple" agents (to avoid any pitfalls in the automation)

I guess I assumed typing in the exact thing as the preferred name was making the selection process simple! No worries, I'll manage. Thanks!
